src/Microsoft.Identity.Client/Platforms/net45/NetDesktopPlatformProxy.cs

```diff
@@ -146,8 +146,13 @@ public override string GetBrokerOrRedirectUri(Uri redirectUri)
         }
 
         /// <inheritdoc />
-        public override string GetDefaultRedirectUri(string clientId)
+        public override string GetDefaultRedirectUri(string clientId, bool useRecommendedRedirectUri = false)
         {
+            if (useRecommendedRedirectUri)
+            {
+                return Constants.DefaultDesktopRedirectUri;
```

Comment:
>Constants.DefaultDesktopRedirectUri; [](start = 23, length = 36)

Perhaps this for non system browser and for system browser it should be localhost?